### ISSUE TITLE
Merge to downstream only at well-defined points

### ISSUE THREAD

**Issue by sheerun** [monterail/guidelines#77]
Do not merge to downstream except with a good reason: upstream API changes affect your branch; your branch no longer merges to upstream cleanly; etc.

Otherwise, the feature that was merged to suddenly contains more than a single (well-separated) change. The many resulting small merges will greatly clutter up history. Anyone who later investigates the history of a file will have to find out whether that merge affected the topic in development. An upstream might even inadvertently be merged into a "more stable" branch. And so on.

In short don't merge develop to feature without good reason.

Reference: http://jk.gs/gitworkflows.html
